client/content_publishing_report/controllers/ContentPublishingReportController.js

```diff
@@ -118,7 +118,9 @@ export function ContentPublishingReportController(
 
         $scope.defaultReportParams = _.cloneDeep($scope.currentParams);
 
-        $scope.group_by = _.cloneDeep($scope.report_groups);
+        $scope.group_by = [...$scope.report_groups, ...$scope.config?.group_by.filter(
```

Comment:
Rather than using the backend to generate this for frontend use, you can use the SuperdeskAPI to get it [directly](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/superdesk-api.d.ts#L2867).

```typescript
import {superdeskApi} from '../../superdeskApi';

superdeskApi.entities.vocabulary.getCustomFieldVocabularies()
```

You would no longer need `group_by` in the `ReportConfigs`, so these changes from the backend can be removed

Reply:
This will also be better placed in `$scope.report_groups`, so `updateGroupOptions` will include these custom vocabularies as well, leaving this line unchanged, `$scope.group_by = _.cloneDeep($scope.report_groups);`